worker.go

```diff
@@ -213,18 +210,22 @@ func (worker *AmqpWorker) Start(signals <-chan os.Signal) {
 
 	handlerEnd := make(chan error, len(worker.Handlers))
 
-	channel := getChannel(conn.Channel)
+	for _, handler := range worker.Handlers {
+		formattedQueueName := formatQueueName(worker.Queue, handler.RoutingKey)
 
-	declareExchange(channel.ExchangeDeclare, worker.Exchange)
+		channel := getChannel(conn.Channel)
 
-	queue := declareQueue(channel.QueueDeclare, worker.Queue, nil)
+		declareExchange(channel.ExchangeDeclare, worker.Exchange)
 
-	setChannelQos(channel.Qos, worker.ChannelPrefetchCount)
+		setChannelQos(channel.Qos, worker.ChannelPrefetchCount)
 
-	for _, handler := range worker.Handlers {
-		bindQueue(channel.QueueBind, worker.Exchange, queue.Name, handler.RoutingKey)
+		declareQueue(channel.QueueDeclare, formattedQueueName, nil)
```

Comment:
These functions can actually fail and will panic. Maybe it is time to remove these panic and handle the errors cleanly ? If one handler fails to initialize, close previously opened ressources ?